src/details/ast.hpp

```diff
@@ -408,33 +432,69 @@ class TableIndex {
      */
     auto index() const -> Expression;
     auto range() const -> minilua::Range;
-    auto raw() const -> ts::Node;
+    auto debug_print() const -> std::string;
+};
+/**
+ * class for field_expression nodes
+ */
+class FieldExpression {
+    using FieldExpTuple =
+        std::tuple<std::shared_ptr<Prefix>, Identifier, minilua::Range, GEN_CAUSE>;
+    enum TupleIndex { TABLE, PROPERTY, RANGE, CAUSE };
+    std::variant<ts::Node, FieldExpTuple> content;
+
+public:
+    explicit FieldExpression(ts::Node);
+    explicit FieldExpression(const Prefix&, const Identifier&, minilua::Range, GEN_CAUSE);
+    /**
+     *
+     * @return a Prefix containing the Identifier for the table
+     */
+    auto table_id() const -> Prefix;
+    /**
+     *
+     * @return an Identifier for a property of the Table
+     */
+    auto property_id() const -> Identifier;
+    auto range() const -> minilua::Range;
+    auto debug_print() const -> std::string;
 };
+using VarDecVariant = std::variant<Identifier, FieldExpression, TableIndex>;
```

Comment:
This should probably be nested in `VariableDeclarator` because the name is not very clear.